src/openvswitch.c

```diff
@@ -224,6 +224,48 @@ write_ovs_protocols(const NetplanOVSSettings* ovs_settings, const gchar* bridge,
     g_string_free(s, TRUE);
 }
 
+static gboolean
+check_ovs_ssl(gchar* target)
+{
+    /* Check if target needs ssl */
+    if (g_str_has_prefix(target, "ssl:") || g_str_has_prefix(target, "pssl:")) {
+        /* Check if SSL is configured in ovs_settings_global.ssl */
+        if (!ovs_settings_global.ssl.ca_certificate || !ovs_settings_global.ssl.client_certificate ||
+            !ovs_settings_global.ssl.client_key) {
+            g_fprintf(stderr, "ERROR: openvswitch bridge controller target '%s' needs SSL configuration, but global 'openvswitch.ssl' settings are not set\n", target);
+            exit(1);
+        }
+        return TRUE;
+    }
+    return FALSE;
+}
+
+static void
+write_ovs_bridge_controller_targets(const NetplanOVSController* controller, const gchar* bridge, GString* cmds)
+{
+    gboolean needs_ssl = FALSE;
+    g_autofree gchar* ssl = "";
+    gchar* target = g_array_index(controller->addresses, char*, 0);
+    needs_ssl |= check_ovs_ssl(target);
+    GString* s = g_string_new(target);
+
+    for (unsigned i = 1; i < controller->addresses->len; ++i) {
+        target = g_array_index(controller->addresses, char*, i);
+        needs_ssl |= check_ovs_ssl(target);
```

Comment:
I know this way just looks nice, but currently this means performing 2 unnecessary strcmp's everytime + other conditionals. It's more of a nitpick, but I think I'd prefer this the 'old-fashioned' way, like `if (!needs_ssl) needs_ssl = check_ovs_ssl(target);`. Since if needs_ssl is TRUE, it means we already checked for the ssl settings existence and don't need to do that again (as we'd throw an error if we didn't).

Reply:
Thanks, that's a good optimization!